lib/netplay/tcp/poll_descriptor_set.h

```diff
@@ -52,69 +53,92 @@ class PollDescriptorSet : public IDescriptorSet
 
 	explicit PollDescriptorSet() = default;
 
-	virtual bool add(SOCKET fd) override
+	virtual bool add(IClientConnection* conn) override
 	{
+		TCPClientConnection* tcpConn = dynamic_cast<TCPClientConnection*>(conn);
+		if (!tcpConn)
+		{
+			ASSERT(false, "Invalid connection type: expected TCPClientConnection");
+			return false;
+		}
+
 		constexpr short evt = EventType == PollEventType::READABLE ? POLLIN : POLLOUT;
+		fds_.emplace_back(pollfd{ tcpConn->getRawSocketFd(), evt, 0 });
+		return true;
+	}
 
-		assert(size_ < MAX_FDS_COUNT);
-		if (size_ >= MAX_FDS_COUNT)
+	virtual bool remove(IClientConnection* conn) override
+	{
+		TCPClientConnection* tcpConn = dynamic_cast<TCPClientConnection*>(conn);
+		if (!tcpConn)
 		{
+			ASSERT(false, "Invalid connection type: expected TCPClientConnection");
 			return false;
 		}
-		fds_[size_++] = { fd, evt, 0 };
+		auto it = std::find_if(fds_.begin(), fds_.end(),
+			[tcpConn](const pollfd& pfd) { return tcpConn->getRawSocketFd() == pfd.fd; });
+		fds_.erase(it);
```

Comment:
Should only call `vector::erase()` if `it != fds_.end()`

Reply:
Nice catch! Will fix it.